libglusterfs/src/mem-pool.c

```diff
@@ -221,17 +231,25 @@ __gf_realloc(void *ptr, size_t size)
     GF_ASSERT(header->magic == GF_MEM_HEADER_MAGIC);
 
     tot_size = __gf_total_alloc_size(size);
-    header = realloc(header, tot_size);
-    if (caa_unlikely(!header)) {
+#ifdef DEBUG
+    struct mem_acct_rec *rec;
+
+    rec = gf_mem_remove_acct_info(header->mem_acct, header);
+#endif
+    tmp = realloc(header, tot_size);
+    if (caa_unlikely(!tmp)) {
+#ifdef DEBUG
+        gf_mem_update_acct_info(rec, header, header->size);
```

Comment:
If we failed to realloc, I'm not sure what we update and with what values.

Reply:
When `realloc()` fails, the original memory block is untouched. From man page:

> The realloc() function returns a pointer to the newly allocated memory, which is suitably aligned for any built-in type, or NULL if the request failed. The returned pointer may be the same as ptr if the allocation was not moved (e.g., there was room to expand the allocation in-place), or different from ptr if the allocation was moved to a new address. If size was equal to 0, either NULL or a pointer suitable to be passed to free() is returned.  **If realloc() fails, the original block is left untouched; it is not freed or moved.**

The update is basically to re-add the memory block to the list. The size difference will be 0, so it won't change anything else.